Code/Framework/AzCore/AzCore/RTTI/BehaviorContextEBusEventRawSignature.inl

```diff
@@ -0,0 +1,234 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+#pragma once
+
+namespace AZ::Internal
+{
+    //////////////////////////////////////////////////////////////////////////
+    template<class EBus, BehaviorEventType EventType, class R, class BusType, class... Args>
+    BehaviorEBusEvent<EBus, EventType, R(BusType*, Args...)>::BehaviorEBusEvent(FunctionPointer functionPointer, BehaviorContext* context)
+        : BehaviorMethod(context)
+        , m_functionPtr(functionPointer)
+    {
+        SetParameters<R>(m_parameters, this);
+        SetParameters<Args...>(&m_parameters[s_startNamedArgumentIndex], this);
+        // optional ID parameter
+        SetBusIdType<s_isBusIdParameter == 1>();
+    }
+
+
+    //////////////////////////////////////////////////////////////////////////
+    template<class EBus, BehaviorEventType EventType, class R, class BusType, class... Args>
+    template<bool IsBusId>
+    inline AZStd::enable_if_t<IsBusId> BehaviorEBusEvent<EBus, EventType, R(BusType*, Args...)>::SetBusIdType()
+    {
+        SetParameters<typename EBus::BusIdType>(&m_parameters[s_startArgumentIndex]);
+    }
+
+    //////////////////////////////////////////////////////////////////////////
+    template<class EBus, BehaviorEventType EventType, class R, class BusType, class... Args>
+    template<bool IsBusId>
+    inline AZStd::enable_if_t<!IsBusId> BehaviorEBusEvent<EBus, EventType, R(BusType*, Args...)>::SetBusIdType()
+    {
+    }
```

Comment:
This can be combined into a single function using `if constexpr`

```suggestion
    template<class EBus, BehaviorEventType EventType, class R, class BusType, class... Args>
    template<bool IsBusId>
    void BehaviorEBusEvent<EBus, EventType, R(BusType*, Args...)>::SetBusIdType()
    {
        if constexpr (IsBusId)
        {
            SetParameters<typename EBus::BusIdType>(&m_parameters[s_startArgumentIndex]);
        }
    }
```

Reply:
Great suggestion, fixed!